src/components/customers/AddWalletToCustomerDialog.tsx

```diff
@@ -0,0 +1,220 @@
+import { forwardRef, RefObject } from 'react'
+import { gql } from '@apollo/client'
+import { useFormik } from 'formik'
+import { object, string, date } from 'yup'
+import styled from 'styled-components'
+
+import { theme } from '~/styles'
+import { Alert, Button, Dialog, DialogRef, Typography } from '~/components/designSystem'
+import { DatePickerField, TextInput, TextInputField } from '~/components/form'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import {
+  CreateCustomerWalletInput,
+  useCreateCustomerWalletMutation,
+  CurrencyEnum,
+} from '~/generated/graphql'
+import { addToast } from '~/core/apolloClient'
+
+gql`
+  mutation createCustomerWallet($input: CreateCustomerWalletInput!) {
+    createCustomerWallet(input: $input) {
+      id
+    }
+  }
+`
+
+export interface AddWalletToCustomerDialogRef extends DialogRef {}
+
+interface AddWalletToCustomerDialogProps {
+  customerId: string
+}
+
+export const AddWalletToCustomerDialog = forwardRef<DialogRef, AddWalletToCustomerDialogProps>(
+  ({ customerId }: AddWalletToCustomerDialogProps, ref) => {
+    const { translate } = useInternationalization()
+    const [createWallet] = useCreateCustomerWalletMutation({
+      onCompleted(res) {
+        if (res?.createCustomerWallet) {
+          addToast({
+            severity: 'success',
+            translateKey: 'text_62d6d5739e4eee96c1afaee8',
+          })
+        }
+      },
+    })
+
+    const formikProps = useFormik<Omit<CreateCustomerWalletInput, 'customerId'>>({
+      initialValues: {
+        expirationDate: undefined,
+        grantedCredits: '',
+        name: '',
+        paidCredits: '',
+        rateAmount: '1.00',
+      },
+      validationSchema: object().shape({
+        expirationDate: date(),
+        grantedCredits: string().required(''),
+        name: string(),
+        paidCredits: string().required(''),
+        rateAmount: string().required(''),
+      }),
+      validateOnMount: true,
+      onSubmit: async (values) => {
+        let res = await createWallet({
+          variables: {
+            input: {
+              customerId,
+              ...values,
+            },
+          },
+          refetchQueries: ['getCustomer'],
+        })
+
+        const { errors } = res
+
+        if (!errors) {
+          ;(ref as unknown as RefObject<DialogRef>)?.current?.closeDialog()
+        }
```

Comment:
You don't need that here, just do it on the onClick after `formikProps.submitForm` 
We use this only if we need to display an error in the modal

Reply:
won't that close the modal if any backend error occcur ? 500 or whatever

Reply:
There is a toast that will appear but yes it will close the modal. This is the behaviour we have everywhere right now, but feel free to challenge it with Mike if you think it's not ok :)

Reply:
Ok got it. Done